### ISSUE TITLE
Why model is not resized to the length of new tokenizer after adding pad token

### ISSUE THREAD

**Issue by rjtshrm** [facebookresearch/llama-recipes#199]
I see that here  a special token is added for the padding. Then why the model is not resized to +1. Also, doing this we also need to fine-tune the embed_tokens and lm_head layer. Or Am I missing something?

https://github.com/facebookresearch/llama-recipes/blob/0b2fa40dba83fd625bc0f1119b0172c0575b5a57/src/llama_recipes/finetuning.py#L125-L129

**Comment by jveronvialard** [comment]
+1 to this question. I have also opened a PR that adds special chat tokens to the tokenizer (in addition to pad token): https://github.com/facebookresearch/llama-recipes/pull/207. A potential problem I see of resizing to just +1 is that it is no longer a multiple of 8 and can lead to degradation of performance (no longer using tensor cores).

**Comment by HamidShojanazeri** [comment]
@rjtshrm thats right we need if using padding, we mostly with SamSum and Oass are not padding and only concatenating thats why it has bot been caught. Are you using a custom dataset?  

```
model.resize_token_embeddings(model.config.vocab_size + 1)
```

**Comment by HamidShojanazeri** [comment]
> +1 to this question. I have also opened a PR that adds special chat tokens to the tokenizer (in addition to pad token): #207. A potential problem I see of resizing to just +1 is that it is no longer a multiple of 8 and can lead to degradation of performance (no longer using tensor cores).

@rjtshrm  That's correct, talk with folks from HF to get the latest on best way to handle this.

**Comment by HamidShojanazeri** [comment]
closing the issue as it's gone stale.
